### PR TITLE
Build cleanup

### PR DESCRIPTION
**Pull request by basil**
See self-review for explanation of changes.

### Testing done

`mvn clean verify`

### CODE REVIEW

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 http://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
+  <extension>
+    <groupId>io.jenkins.tools.incrementals</groupId>
+    <artifactId>git-changelist-maven-extension</artifactId>
+    <version>1.7</version>
+  </extension>
+</extensions>
```
Comment (basil) [inline .mvn/extensions.xml:7]: Enables [incremental builds](https://github.com/jenkinsci/incrementals-tools) in CI.

File: .mvn/maven.config [hunk 2]
```diff
@@ -0,0 +1,2 @@
+-Pconsume-incrementals
+-Pmight-produce-incrementals
```
Comment (basil) [inline .mvn/maven.config:2]: Enables [incremental builds](https://github.com/jenkinsci/incrementals-tools) in CI.

File: pom.xml [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
   <groupId>org.jvnet.hudson.plugins</groupId>
   <artifactId>thinBackup</artifactId>
-  <version>1.19-SNAPSHOT</version>
+  <version>${revision}${changelist}</version>
```
Comment (basil) [inline pom.xml:14]: Enables [incremental builds](https://github.com/jenkinsci/incrementals-tools) in CI.

File: pom.xml [hunk 4]
```diff
@@ -61,17 +61,18 @@
   </developers>
 
   <scm>
-    <connection>scm:git:https://github.com/jenkinsci/thin-backup-plugin.git</connection>
-    <developerConnection>scm:git:https://git@github.com/jenkinsci/thin-backup-plugin.git</developerConnection>
+    <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
+    <developerConnection>scm:git:git@github.com:${gitHubRepo}.git</developerConnection>
     <tag>${scmTag}</tag>
-    <url>https://github.com/jenkinsci/thin-backup-plugin</url>
```
Comment (basil) [inline pom.xml:67]: Enables [incremental builds](https://github.com/jenkinsci/incrementals-tools) in CI.

File: pom.xml [hunk 5]
```diff
@@ -61,17 +61,18 @@
...
+    <url>https://github.com/${gitHubRepo}</url>
   </scm>
 
   <properties>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (basil) [inline pom.xml:71]: Defined in the plugin parent POM.

File: pom.xml [hunk 6]
```diff
@@ -61,17 +61,18 @@
...
+    <revision>1.19</revision>
+    <changelist>-SNAPSHOT</changelist>
```
Comment (basil) [inline pom.xml:72]: Enables [incremental builds](https://github.com/jenkinsci/incrementals-tools) in CI.

File: pom.xml [hunk 7]
```diff
@@ -61,17 +61,18 @@
...
     <jenkins.version>2.401.3</jenkins.version>
+    <gitHubRepo>jenkinsci/thin-backup-plugin</gitHubRepo>
```
Comment (basil) [inline pom.xml:74]: Enables [incremental builds](https://github.com/jenkinsci/incrementals-tools) in CI.

File: pom.xml [hunk 8]
```diff
@@ -61,17 +61,18 @@
...
     <spotless.check.skip>false</spotless.check.skip>
-    <surefire.forkNumber>1</surefire.forkNumber>
```
Comment (basil) [inline pom.xml:74]: Does not seem necessary; I successfully ran the test suite in parallel locally without issues.

File: pom.xml [hunk 9]
```diff
@@ -91,11 +92,6 @@
       <groupId>io.jenkins.plugins</groupId>
       <artifactId>ionicons-api</artifactId>
     </dependency>
-    <dependency>
-      <groupId>org.junit.jupiter</groupId>
-      <artifactId>junit-jupiter</artifactId>
-      <scope>test</scope>
-    </dependency>
```
Comment (basil) [inline pom.xml:98]: Already defined in the plugin parent POM.

File: pom.xml [hunk 10]
```diff
@@ -115,21 +111,4 @@
...
-          <source>11</source>
-          <target>11</target>
-        </configuration>
-      </plugin>
-      <plugin>
-        <groupId>com.github.spotbugs</groupId>
-        <artifactId>spotbugs-maven-plugin</artifactId>
-      </plugin>
-    </plugins>
-  </build>
```
Comment (basil) [inline pom.xml:134]: Already defined in the plugin parent POM.

**Comment by StefanSpieker** [comment]
Does this change anything when I do a release? I read the documentation and it seems to have no downsides I can see.

**Comment by basil** [comment]
No, this doesn't change anything about the manual release process. It does make it easier to adopt automatic plugin release in the future, should you choose to do so, as one of the prerequisites for automated plugin release is to enable incrementals.

**Review by StefanSpieker — APPROVED** [review]

**Pull request merged**